sdks/python/apache_beam/runners/interactive/display/pcoll_visualization.py

```diff
@@ -238,31 +322,57 @@ def _display_dive(self, data, update=None):
       display(HTML(html))
 
   def _display_overview(self, data, update=None):
+    if (not data.empty and self._include_window_info and
+        all(column in data.columns
+            for column in ('event_time', 'windows', 'pane_info'))):
+      data = data.drop(['event_time', 'windows', 'pane_info'], axis=1)
+
     gfsg = GenericFeatureStatisticsGenerator()
     proto = gfsg.ProtoFromDataFrames([{'name': 'data', 'table': data}])
     protostr = base64.b64encode(proto.SerializeToString()).decode('utf-8')
     if update:
       script = _OVERVIEW_SCRIPT_TEMPLATE.format(
-          display_id=update, protostr=protostr)
+          display_id=update._overview_display_id, protostr=protostr)
       display_javascript(Javascript(script))
     else:
       html = _OVERVIEW_HTML_TEMPLATE.format(
           display_id=self._overview_display_id, protostr=protostr)
       display(HTML(html))
 
   def _display_dataframe(self, data, update=None):
-    if update:
-      table_id = 'table_{}'.format(update)
-      html = _DATAFRAME_PAGINATION_TEMPLATE.format(
-          dataframe_html=data.to_html(notebook=True, table_id=table_id),
-          table_id=table_id)
-      update_display(HTML(html), display_id=update)
+    table_id = 'table_{}'.format(
+        update._df_display_id if update else self._df_display_id)
+    columns = [{
+        'title': ''
+    }] + [{
+        'title': str(column)
+    } for column in data.columns]
+    format_window_info_in_dataframe(data)
+    rows = data.applymap(lambda x: str(x)).to_dict('split')['data']
```

Comment:
What is happening in here in the next few lines?

Reply:
Could you explain that in a comment there?

Reply:
Added the comments.